src/odemis/gui/model/__init__.py

```diff
@@ -880,6 +880,11 @@ def __init__(self, name, stage=None, stream_classes=None, fov_hw=None):
         else:
             self.stream_classes = stream_classes
         self._stage = stage
+        # TODO fix it
+        if isinstance(self.stream_classes, tuple):
```

Comment:
Agree, that looks weird. Probably the right way to deal with it is to make the code checking `.stream_classes` (eg, the stream controller) aware of the DataProjection. If it's a DataProjection, it should look at its stream to decide whether the class is correct.